qa/tasks/cephfs/xfstests_dev.py

```diff
@@ -0,0 +1,171 @@
+import logging
+from StringIO import StringIO
+from tasks.cephfs.cephfs_test_case import CephFSTestCase
+
+log = logging.getLogger(__name__)
+
+
+class XFSTestsDev(CephFSTestCase):
+    def setUp(self):
+        CephFSTestCase.setUp(self)
+        self.prepare_xfstests_dev()
+
+    def prepare_xfstests_dev(self):
+        self.get_repo()
+        self.get_test_and_scratch_dirs_ready()
+        self.install_deps()
+
+        # NOTE: On teuthology machines it's necessary to run "make" as
+        # superuser since the repo is cloned somewhere in /tmp.
+        self.mount_a.client_remote.run(args=['sudo', 'make'],
+                                       cwd=self.repo_path, stdout=StringIO(),
+                                       stderr=StringIO())
+
+        self.mount_a.client_remote.run(args=['sudo', 'make', 'install'],
+                                       cwd=self.repo_path, omit_sudo=False,
+                                       stdout=StringIO(), stderr=StringIO())
+
+        self.create_reqd_users()
+        self.write_local_config()
+
+    def get_repo(self):
+        """
+        Clone xfstests_dev repository. If already present, update it.
+        """
+        from os import getcwd
+        from os import stat
+        from os.path import join
+        from subprocess import call
+
+        # TODO: make sure that repo is not cloned for every test. it should
+        # happen only once.
+        repourl = 'git://git.kernel.org/pub/scm/fs/xfs/xfstests-dev.git'
+        self.repo_path = join(self.mount_a.client_remote.mkdtemp(),
+                              'xfstests-dev')
+        self.mount_a.run_shell('%s %s %s' % ('git clone', repourl,
+                                             self.repo_path))
```

Comment:
We can't be hitting `git.kernel.org` for every test we run. The right thing to do here is to export a tarball of some tagged version of xfstests-dev and hand that off to @djgalloway to put somewhere on ceph.com. Then just pull that down in the tests and build.

Reply:
I'm guessing that's why we have xfs-tests.git on git.ceph.com: http://git.ceph.com/?p=xfstests.git;a=summary

Shall I just add xfstests-dev?

Reply:
Works for me!

@rishabh-d-dave see `qa/tasks/restart.py` for how we download a tarball from that remote. There's no reason to clone the remote.

Reply:
http://git.ceph.com/?p=xfstests-dev.git;a=summary

Reply:
@rishabh-d-dave

Reply:
Cloning from `git.ceph.com`, now. Putting this to test on teuthology next.

Would've been nicer if I knew this beforehand since I had problems with cloning from `git.kernel.org`.

Reply:
@djgalloway Thanks!